lib/mobility-core/src/Kernel/External/Notification/FCM/Types.hs

```diff
@@ -440,8 +442,12 @@ instance (FromJSON a) => FromJSON (FCMData a) where
       <*> o .: "entity_ids"
       <*> (o .: "entity_data" >>= parseNotificationJson)
       <*> (o .: "notification_json" >>= parseNotificationJson)
-      <*> (o .:? "driver_notification_payload" >>= (maybe (pure Nothing) parseNotificationJson))
+      <*> (o .:? "notification_id" >>= maybe (pure Nothing) parseNotificationId) -- need to add this
```

Comment:
Just this much should be ok to decode the optional notification_id field.
`<*> o .:? "notification_id"`